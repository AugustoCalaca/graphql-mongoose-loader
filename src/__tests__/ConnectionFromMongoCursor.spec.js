// @flow'

import {
  clearDbAndRestartCounters,
  connectMongooseAndPopulate,
  disconnectMongoose,
  createUser,
  createGroup,
} from '../../test/helpers';

import connectionFromMongoCursor from '../ConnectionFromMongoCursor';
import UserModel from '../../test/fixtures/UserModel';

beforeAll(connectMongooseAndPopulate);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it('should return connection from mongo cursor', async () => {
  // 2 users without groups
  const userA = await createUser();
  const userB = await createUser();

  // 2 users with groups
  const GroupA = await createGroup();
  const GroupB = await createGroup();

  await createUser({ group: GroupA });
  await createUser({ group: GroupB });

  const cursor = UserModel.find();

  const context = { randomProperty: 2 };
  const loader = jest.fn();
  loader.mockReturnValue('user');

  const args = { first: 2 };

  const result = await connectionFromMongoCursor({ cursor, context, args, loader });

  expect(result).toMatchSnapshot();
  expect(loader).toHaveBeenCalledTimes(2);
  expect(loader.mock.calls[0]).toEqual([context, userA._id]);
  expect(loader.mock.calls[1]).toEqual([context, userB._id]);
});
