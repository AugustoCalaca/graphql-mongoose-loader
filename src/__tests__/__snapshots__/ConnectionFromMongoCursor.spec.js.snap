// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return connection from mongo cursor 1`] = `
Object {
  "count": 4,
  "edges": Array [
    Object {
      "cursor": "bW9uZ286MA==",
      "node": "user",
    },
    Object {
      "cursor": "bW9uZ286MQ==",
      "node": "user",
    },
  ],
  "endCursorOffset": 2,
  "pageInfo": Object {
    "endCursor": "bW9uZ286MQ==",
    "hasNextPage": true,
    "hasPreviousPage": false,
    "startCursor": "bW9uZ286MA==",
  },
  "startCursorOffset": 0,
}
`;
